install.ps1

```diff
@@ -1,429 +1,931 @@
+        }
+
+        # Remove default packages
+        $nodes = $configXml.SelectNodes('//config/packages/package')
+        foreach($node in $nodes) {
+            $node.ParentNode.RemoveChild($node) | Out-Null
+        }
+
+        # Add environment variables
+        $envs = $configXml.SelectSingleNode('//envs')
+        $newXmlNode = $envs.AppendChild($configXml.CreateElement("env"));
+        $newXmlNode.SetAttribute("name", "VM_COMMON_DIR");
+        $newXmlNode.SetAttribute("value", $vmCommonDirText.text);
+        $newXmlNode = $envs.AppendChild($configXml.CreateElement("env"));
+        $newXmlNode.SetAttribute("name", "TOOL_LIST_DIR");
+        $newXmlNode.SetAttribute("value", $toolListDirText.text);
+        $newXmlNode = $envs.AppendChild($configXml.CreateElement("env"));
+        $newXmlNode.SetAttribute("name", "TOOL_LIST_SHORTCUT");
+        $newXmlNode.SetAttribute("value", $toolListShortCutText.text);
+        $newXmlNode = $envs.AppendChild($configXml.CreateElement("env"));
+        $newXmlNode.SetAttribute("name", "RAW_TOOLS_DIR");
+        $newXmlNode.SetAttribute("value", $rawToolsDirText.text);
+
+        # Add selected packages
+        $packages = $configXml.SelectSingleNode('//packages')
+        foreach($package in $selectedPackagesBox.Items) {
+            $newXmlNode = $packages.AppendChild($configXml.CreateElement("package"));
+            $newXmlNode.SetAttribute("name", $package);
+        }
+    } else {
+        Write-Host "[+] Cancel pressed, stopping installation..."
+        Start-Sleep 3
+        exit 1
+    }
 
-Write-Host "[+] Installing Boxstarter"
-$rc = installBoxStarter
-if ( -Not $rc ) {
-  Write-Host "[ERR] Failed to install BoxStarter`n" -ForegroundColor Red
-  Read-Host  "`tPress ANY key to continue..."
-  exit
+    ################################################################################
+    ## END GUI
+    ################################################################################
 }
 
-# Boxstarter options
-if ($norestart) {
-  $Boxstarter.RebootOk = $false
-} else {
-  $Boxstarter.RebootOk = $true # Allow reboots?
+# Save the config file
+Write-Host "[+] Saving configuration file..."
+$configXml.save($configPath)
+$configXml.save((Join-Path ${Env:VM_COMMON_DIR} "config.xml"))
+
+# Parse config and set initial environment variables
+Write-Host "[+] Parsing configuration file..."
+foreach ($env in $configXml.config.envs.env) {
+    $path = [Environment]::ExpandEnvironmentVariables($($env.value))
+    Write-Host "`t[+] Setting %$($env.name)% to: $path" -ForegroundColor Green
+    [Environment]::SetEnvironmentVariable("$($env.name)", $path, "Machine")
 }
-$Boxstarter.NoPassword = $false # Is this a machine with no login password?
-$Boxstarter.AutoLogin = $true # Save my password securely and auto-login after a reboot
-Set-BoxstarterConfig -NugetSources "https://www.myget.org/F/fireeye/api/v2;https://chocolatey.org/api/v2"
+refreshenv
```

Comment:
Getting this error here:
```
Join-Path : Cannot bind argument to parameter 'Path' because it is null.
At C:\Users\user\Desktop\flare-vm\install.ps1:812 char:28
+ $configXml.save((Join-Path ${Env:VM_COMMON_DIR} "config.xml"))
+                            ~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Join-Path], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.JoinPathCom
   mand

```

Reply:
maybe there's a sleep needed as I see the env var setting after this?